### ISSUE TITLE
too many load balancers required?

### ISSUE THREAD

**Issue by mindnuts** [awslabs/ecs-refarch-service-discovery#8]
Hi...if i understood it correctly for this reference architecture, one would require one ELB infront of each microservice. Imagine if you have stack with around 20 microservices. You would end up with fixed costs of $216 x 20 = $4320 plus any [data transfer charges](https://aws.amazon.com/elasticloadbalancing/pricing/) for ELB alone!

Is this correct, if yes can the number of ELBs be somehow reduced?

**Comment by kostyrevaa** [comment]
@mindnuts Hi. I'm new to AWS. Could you please elaborate on this $216? Where this number comes from?

**Comment by mindnuts** [comment]
@all If anyone is following this thread, there is another [approach](https://github.com/awslabs/service-discovery-ecs-dns) from AWS labs.

TLDR; You would need one ALB/ELB per internet facing microservice in this approach. Its still high but its something.

--M

**Comment by ghost** [comment]
Why not using Consul for Service Discovery ? No LB needed internaly.
